storage/shard.go

```diff
@@ -732,7 +777,13 @@ func (s *dbShard) Write(
 	// If no entry and we are not writing new series asynchronously
 	if !writable && !opts.writeNewSeriesAsync {
 		// Avoid double lookup by enqueueing insert immediately
-		result, err := s.insertSeriesAsyncBatched(id, dbShardInsertAsyncOptions{})
+		result, err := s.insertSeriesAsyncBatched(id, tags, dbShardInsertAsyncOptions{
+			hasPendingIndexing: reverseIndexWriteFn != nil,
+			pendingIndex: dbShardPendingIndex{
+				timestamp: timestamp,
+				fn:        reverseIndexWriteFn,
```

Comment:
Yeah this would be nicer if you just pass a `databaseIndex` interface here rather than a plain fn, much more flexible in the future if there are other methods we need to call on the index.

Reply:
Can avoid entirely here because of your other point